packages/kit/src/core/sync/create_manifest_data/index.spec.js

```diff
@@ -222,6 +222,8 @@ test('sorts routes correctly', () => {
 		'c/b[x]',
 		'c/foo',
 		'd/e',
+		'd/[[optional=match]]/e',
+		'd/[[optional]]/e',
```

Comment:
Found one more case which didn't sort correctly, for which I had to bring back the "second pass"-logic from my other branch. It doesn't complicate the logic in the sense of explaining it, only the implementation.

Reply:
Ah yes, I didn't think this through enough, one would have to be required - yes we can revert it then, and finally merge.